File: node-ui/classes/Database/TwitterDailyStatsAccounts.js
const { DataTypes } = require('sequelize');
const sequelize = require('./base');
const TwitterAccounts = require('./TwitterAccounts');

const TwitterDailyStatsAccounts = sequelize.define('twitter_dailystats_accounts', {
    handle: {
        type: DataTypes.STRING(255),
        allowNull: false,
        primaryKey: true,
        references: {
            model: TwitterAccounts,
            key: 'handle',
        },
        onDelete: 'CASCADE'
    },
    date: {
        type: DataTypes.DATEONLY,
        allowNull: false,
        primaryKey: true
    },
    followers_count: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    fetched_tweets_count: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    tot_likes: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    tot_retweets: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    tot_replies: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    avg_len: {
        type: DataTypes.DECIMAL(11, 2),
        allowNull: false
    },
    avg_likes: {
        type: DataTypes.DECIMAL(11, 2),
        allowNull: false
    },
    avg_retweets: {
        type: DataTypes.DECIMAL(11, 2),
        allowNull: false
    },
    avg_replies: {
        type: DataTypes.DECIMAL(11, 2),
        allowNull: false
    },
    avg_sentiment: {
        type: DataTypes.ENUM('Positive', 'Negative', 'Neutral', 'Null'),
        allowNull: false
    }
});

TwitterDailyStatsAccounts.belongsTo(TwitterAccounts, {foreignKey: 'handle', targetKey: 'handle', onDelete: 'CASCADE'});

module.exports = TwitterDailyStatsAccounts

File: py-crawler/classes/databases/twitter_dailystats_accounts.py
from sqlalchemy import Column
from sqlalchemy import String, Integer, DECIMAL, ForeignKey

from .base import Base

# twitter_dailystats_accounts table.
# Author: Antonio Scardace

class TwitterDailyStatsAccounts(Base):

    __tablename__ = 'twitter_dailystats_accounts'

    handle = Column('handle', String(255), ForeignKey('twitter_accounts.handle', ondelete='CASCADE'), nullable=False, primary_key=True)
    date = Column('date', String(64), nullable=False, primary_key=True)
    followers_count = Column('followers_count', Integer, nullable=False)
    fetched_tweets_count = Column('fetched_tweets_count', Integer, nullable=False)
    tot_likes = Column('tot_likes', Integer, nullable=False)
    tot_retweets = Column('tot_retweets', Integer, nullable=False)
    tot_replies = Column('tot_replies', Integer, nullable=False)
    avg_len = Column('avg_len', DECIMAL(10, 2), nullable=False)
    avg_likes = Column('avg_likes', DECIMAL(10, 2), nullable=False)
    avg_retweets = Column('avg_retweets', DECIMAL(10, 2), nullable=False)
    avg_replies = Column('avg_replies', DECIMAL(10, 2), nullable=False)
    avg_sentiment = Column('avg_sentiment', String, nullable=False)

    def __init__(self, handle: str, date):
        self.handle = handle
        self.date = date